client/state/wp-data/calypso-wp-data-store.js

```diff
@@ -0,0 +1,20 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { fetchPreferences, setPreference, savePreference } from 'state/preferences/actions';
+import { getPreference, isFetchingPreferences } from 'state/preferences/selectors';
+
+export default {
+	useCalypsoStore: true,
+	selectors: {
+		getPreference,
+		isFetchingPreferences,
```

Comment:
can you put them in their own file and import with the asterisk? This is what we do in Gutenberg. We make all selectors and actions public by design.

Reply:
it's more about the bundle size than the list of imports. if I'm going to use `getPreference` in a plugin I don't want to pull in all of the hundreds of selectors, reducers, and action creators from Calypso

Reply:
If we use `@wordpress/data` interfaces to access the Calypso Redux store, the interface won't be one giant store with hundreds of actions and selectors. It will be divided into many `@wordpress/data` stores: `calypso/preferences`, `calypso/posts`, `calypso/plugins`, ...

The client will access the stores using the `select` and `dispatch` functions or using the `withSelect` and `withDispatch` HoCs:
```js
withSelect( select => ( {
  pluginsList: select( 'calypso/plugins' ).getWPorgPlugins( { category: 'writing' } )
} ) );
```

That doesn't necessarily mean that `calypso/plugins` will be a separate Redux store. `calypso/plugins` is an interface that has some contract (actions and selectors) and the implementation has a lot of freedom on how exactly to fulfill that contract.

It's an open question if these interfaces make code splitting possible or not. I just wanted to point out that the "file with a giant list of all of our selectors" scenario is very unlikely to happen.

Reply:
maybe I'm mistaken but I believe that the purpose of this PR was to explore a wrapper of all of Calypso's data code behind _one_ store vs. rewriting each separate area of state into its own store, meaning that the purpose was to explore "one giant store with hundreds of actions and selectors."

?

Reply:
Personally, I'd like to use this as an opportunity to break up the monolith of state we have in Calypso, but I'd like for us to be deliberate about where we draw those lines of separation. I think it could make sense for a collection of selectors to use the same store, if it's similar data. In the end, I'd like to see Calypso using a small number of stores where the division lines are clear. It may also make sense for more than one `wp.data` store to use the same state tree within Calypso, for performance reasons (i.e. we don't want to have 50+ redux stores)

Reply:
In a9a0ebd I moved this to a store named 'calypso/preferences' within a `calypso-stores` directory with an index. I think this makes sense and will be much easier to split out from the shared state as well.